library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -323,19 +322,21 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
         continue;
       }
 
+      // If we've reached this point then there is data to process; flag that work has been done.
+      captionDataProcessed = true;
+
       boolean previousCaptionValid = captionValid;
       captionValid = (ccHeader & CC_VALID_FLAG) == CC_VALID_FLAG;
       if (!captionValid) {
         if (previousCaptionValid) {
           // The encoder has flipped the validity bit to indicate captions are being turned off.
           resetCueBuilders();
-          captionDataProcessed = true;
         }
         continue;
       }
 
-      // If we've reached this point then there is data to process; flag that work has been done.
-      captionDataProcessed = true;
+      boolean repeatedControlPossible = repeatableControlSet;
+      repeatableControlSet = false;
```

Comment:
Should these lines go above the block that handles invalid captions? I doubt it's correct to de-duplicate repeated commands that have invalid data in between. Not sure though :).

Reply:
I was thinking it should go after the zero check and before the validity check. In case captions are toggled off by flipping the validity bit after a repeatable command, and then when captions are toggled on again the first piece of data happens to be the same repeatable command. Seems like an edge case, but may as well handle it given it's just moving the lines up a little?

Reply:
I will move this up as part of the merge.